pkg/cloud/scalewayprovider.go

```diff
@@ -0,0 +1,199 @@
+package cloud
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/kubecost/opencost/pkg/clustercache"
+	"github.com/kubecost/opencost/pkg/util"
+
+	"github.com/kubecost/opencost/pkg/log"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type ScalewayPricing struct {
+	NodesInfos map[string]*instance.ServerType
+	PVCost     float64
+}
+
+type Scaleway struct {
+	Clientset               clustercache.ClusterCache
+	Config                  *ProviderConfig
+	Pricing                 map[string]*ScalewayPricing
+	DownloadPricingDataLock sync.RWMutex
+}
+
+func (c *Scaleway) DownloadPricingData() error {
+	c.DownloadPricingDataLock.Lock()
+	defer c.DownloadPricingDataLock.Unlock()
+
+	// TODO wait for an official Pricing API from Scaleway
+	// Let's use a static map and an old API
+
+	if len(c.Pricing) != 0 {
+		// Already initialized
+		return nil
+	}
+
+	// PV pricing per AZ
+	pvPrice := map[string]float64{
+		"fr-par-1": 0.00011,
+		"fr-par-2": 0.00011,
+		"fr-par-3": 0.00032,
+		"nl-ams-1": 0.00008,
+		"nl-ams-2": 0.00008,
+		"pl-waw-1": 0.00011,
+	}
+
+	c.Pricing = make(map[string]*ScalewayPricing)
+
+	client, err := scw.NewClient(scw.WithAuth("SCWXXXXXXXXXXXXXXXXX", "00000000-0000-0000-0000-000000000000"), scw.WithDefaultProjectID("00000000-0000-0000-0000-000000000000"))
```

Comment:
Not for the author, but just a note that we can pick this up on our side. Historically if these aren't public (AWS is, but GCP/Azure are not), we don't mind popping in a key we own with extremely locked down permissions, or working with the Scaleway folks to get a key so that users can query this.

Reply:
As stated above, this might be useful for the future pricing API, for now it's just a workaround for the scw client (added a comment too)